### DISCUSSION TITLE
Can't make it work

### DISCUSSION THREAD

**Discussion by Purplen** [MarkusJx/autobet#28] [Issues] [answered]
Hello!

I just downloaded today the latest release, but can't get it to work properly.
I can tell by the logs that it finds the right elements, but simply missclicks. Like the mouse is  stuck to the right edge of the window.
Also when I try to start through the webserver, it says that the game is not running although in the app it says it does, and tries to do something.
I tried avoiding the missclicks by setting it to controller mode, but then again it doesn't do anything. (and yes, I have the Scp installed, just a the wiki instructed)
I got so desperate that I'm doing something wrong I wanted to check the demonstration video and got Rickrolled.

I'm wondering if It was a problem that you fixed ages ago but didn't make a new release to it yet or something, as I see you keep updating the project, but the latest release was in march.

Anyway, got any idea what do I do wrong?
Thanks in advance,

PS:
I attached some pictures by the debug, might help.
[autobet_debug.zip](https://github.com/MarkusJx/autobet/files/6642966/autobet_debug.zip)

**Comment by MarkusJx** [comment, marked as answer]
Hey,

if your mouse isn't moving, that's in general not an issue with autobet.
I've seen this before and it was caused by some gaming-mouse-whatever-software being installed.
So if you've got some sort of this software installed, just try without it (uninstall, stopping it may or may not work as you can't really stop that stuff from being annoying). Maybe. Most of those tools are shit anyway.

But this could also be caused by the mouse not being captured correctly by the game (it's a R* game, so yeah, that stuff happens),
just tab out of the game and tab back in and see if it works. You may also wanna try restarting the game, this might help as well.

With the webserver not-working-thing, I'll have to look into that, but it should work (obviously it doesn't, duh).

How did you install ScpVBus, the vjoystick website is still down, right? Whatever. You somehow got it to work (you can't even select the controller navigation without ScpVBus being installed, no worries).

And yes, I've added the video links just to fuck with people. I actually do have a demonstration video (which could be accessed on the old website and is still linked in the wiki), but watching it, I've noticed how shitty that thing is. I mean, I can't really show how this program works, if it...well, works. You just press a few buttons and that's it. Not much to see here.

I hope this helps and let me know if those "fixes" worked.
Have a nice day

**Edit:** *Solution in comments*

**Reply by Purplen** [reply]
It took me some time but I figured it out.
The problem was my monitor setup, After I disconnected every other monitor except one, it started to work wonderfully.

Also for the ScpVBus, I just found an unofficial mirror, so I am able to select it, I checked the Scp logs, and it works ( I mean Autobet does controls a virtual controller), just the game doesn't really takes any input from it. But I've seen it navigating the bet menu in the wrong way once. So I assume that it's a problem with my Scp settings or the unofficial install itself.

About not having a demonstration video, what you wrote 100% makes sense, the tool is pretty self explanatory. I wouldn't have tried to watch it if I didn't get stuck either. I just wrote that part to let you know, you achieved a double rickroll lmao.

Thanks for the help anyway, love the tool, great work!

**Reply by MarkusJx** [reply]
Okay, that's weird, it usually works with multiple screens, but maybe there is something wrong with those windows APIs, who tf knows... I'll look into it. Definitely would've been easier to keep using those python packages...whatever.

Anyways, good to know everything works (somewhat) and I'm always happy to help.

If you have any further questions, feel free to start a new thread or open a new issue, I'm sure you know how this GitHub-thing works.

:)

**Reply by MarkusJx** [reply]
I don't know if you get a notification about this, but I don't really care, I'll write this here anyway.

I just fixed the mouse coordinates for multi screen setups (or it should be fixed).
Looks like screens on windows are allowed to have some sort of offset, which, when not subtracted from the coordinate to click on, would cause the cursor to be stuck in the right side.

I don't know if this works, as is was already working on my system(s), so you may wanna try the latest beta build.

**Reply by Purplen** [reply]
Oh sweet!
I'll check it out this afternoon. You're super fast replying and fixing stuff, I really appreciate it.

**Reply by Purplen** [reply]
Now it works with my monitor setup as well.
But I’m still not able to start or stop through the web interface. It still says that the game isn’t running which is kinda funny. But at least I don’t have to disable my other screens so I can watch a movie. This night I’ll dive into the ScpVBus stuff and try making it work.

**Reply by MarkusJx** [reply]
I actually don't recommend using the controller simulation thing as it is pretty slow (you'll make more money when using mouse controls, those need way smaller timeouts) unless really required. This feature was originally intended to be used with game streaming from an XBox, as this specifically requires a controller and can't be controlled using the mouse. So...I'd stick with the mouse controls, as the controller thing doesn't really have any advantages.

As for the website not working, I recall it working, did you wait some time before trying to start the game. It may take some time for the web code to register that the game is running as those 'game running' messages are only sent every ~10 seconds (actually, it's 12 seconds, don't ask me why), thus, in the worst case, you'll have to wait 10(12) seconds after opening the web page before the start button will work.

To make sure it really works, I'll test the web page again myself and add a info text always showing if the game is running on your machine in the hope of making the start/stop mechanics less confusing and less
![Kinda_Funny_logo_on_black-1920](https://user-images.githubusercontent.com/28785953/122167896-df71ab80-ce7b-11eb-9786-e7b3f83c7828.jpg)

(I'm sorry, I had to do that)

**Reply by MarkusJx** [reply]
Game running status message added. Check the latest beta build to test it.

**Comment by MarkusJx** [comment]
The web thing should be working now with the latest beta build, thank you for noticing.
